### PR TITLE
added solutiom

### PR DESCRIPTION
**Pull request by MarchD**
[DEMO LINK](https://MarchD.github.io/react_static-list-of-posts/)

### CODE REVIEW

**Review by troospieler — APPROVED** [review]
good job

File: src/components/Comment/Comment.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      <a href={`mailto:${email}`}>
+        {email}
+      </a>
+    </p>
+    <p className="comment__name">{name}</p>
+    <p>{body}</p>
+  </div>
+);
+
+Comment.propTypes = TypeOfComment.isRequired;
```
Comment (aionova) [inline src/components/Comment/Comment.js:17]:
```suggestion
Comment.propTypes = {
  name: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
```
If you've destructured comment object and pass separate props, you need to describe their types separately.

File: src/components/Post/Post.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export const Post = ({ body, title, comments, user }) => (
+  <div className="post">
+    <User user={user} />
+    <h2 className="post__title">{title}</h2>
+    <div className="post__body">{body}</div>
+    <CommentList comments={comments} />
+  </div>
+);
+
+Post.propTypes = TypeOfPost.isRequired;
```
Comment (aionova) [inline src/components/Post/Post.js:18]: same here

File: src/components/User/User.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export const User = ({ user }) => (
+  <div className="post__author">
+    <span className="post__author-name">{user.name}</span>
+    <a href={`mailto:${user.email}`}>{user.email}</a>
+    <span>{user.address.city}</span>
+  </div>
+);
+
+User.propTypes = TypeOfUser.isRequired;
```
Comment (aionova) [inline src/components/User/User.js:12]:
```suggestion
User.propTypes = {
  user: TypeOfUser.isRequired,
}
```

**Review by aionova — CHANGES_REQUESTED** [review]
Good start!

File: src/components/CommentList/CommentList.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+export const CommentList = ({ comments }) => (
+  <div className="comments">
+    {comments.map(comment => (
+      <Comment {...comment} key={comment.id} />
+    ))
+    }
+  </div>
+);
+
+CommentList.propTypes = TypeOfCommentsList.isRequired;
```
Comment (aionova) [inline src/components/CommentList/CommentList.js:15]:
```suggestion
CommentList.propTypes = {
  comments: TypeOfCommentsList.isRequired,
};
```

File: src/components/PostList/PostList.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+export const PostList = ({ posts }) => (
+  <div className="posts">
+    {posts.map(post => (
+      <Post {...post} key={post.id} />
+    ))}
+  </div>
+);
+
+PostList.propTypes = TypeOfPostsList.isRequired;
```
Comment (aionova) [inline src/components/PostList/PostList.js:14]:
```suggestion
PostList.propTypes = {
  posts: TypeOfPostsList.isRequired,
};
```

**Review by aionova — APPROVED** [review]
Good job!

File: src/components/CommentList/CommentList.js (hunk 4 above)
Comment (MarchD) [reply src/components/CommentList/CommentList.js:15]: 😞

**Pull request closed without merging**